tests/subsys/storage/flash_map/src/main.c

```diff
@@ -98,9 +99,70 @@ void test_flash_area_get_sectors(void)
 
 }
 
+void test_flash_area_check_int_sha256(void)
+{
+	/* echo $'0123456789abcdef\nfedcba9876543210' > tst.sha
+	 * hexdump tst.sha
+	 */
+	uint8_t tst_vec[] = { 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+			      0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66,
+			      0x0a, 0x66, 0x65, 0x64, 0x63, 0x62, 0x61, 0x39,
+			      0x38, 0x37, 0x36, 0x35, 0x34, 0x33, 0x32, 0x31,
+			      0x30, 0x0a };
+	/* sha256sum tst.sha */
+	uint8_t tst_sha[] = { 0xc6, 0xb6, 0x7c, 0x46, 0xe7, 0x2e, 0x14, 0x17,
+			      0x49, 0xa4, 0xd2, 0xf1, 0x38, 0x58, 0xb2, 0xa7,
+			      0x54, 0xaf, 0x6d, 0x39, 0x50, 0x6b, 0xd5, 0x41,
+			      0x90, 0xf6, 0x18, 0x1a, 0xe0, 0xc2, 0x7f, 0x98 };
+	const struct flash_area *fa;
+	struct flash_area_check fac = { NULL, 0, -1, NULL, 0 };
+	uint8_t buffer[16];
+	int rc;
+
+	rc = flash_area_open(FLASH_AREA_ID(image_1), &fa);
+	zassert_true(rc == 0, "flash_area_open() fail, error %d\n", rc);
+	rc = flash_area_erase(fa, 0, fa->fa_size);
+	zassert_true(rc == 0, "Flash erase failure (%d), error %d\n", rc);
+	rc = flash_area_write(fa, 0, tst_vec, 40);
```

Comment:
This is failing when add a value not multiple from 8 on qemu_x86. It fail with sizeof(tst_vec) => 0x22 (34 dec).
Keeped 40 to pass test and raise problem.

Reply:
Why don't you just set test vector to contain multiple of 8 characters?

Reply:
And by rounding the value to 40 just like that, aren't you actually crossing the tst_vec boundary and wandering into stack area that is not part of tst_vec?

Reply:
> Why don't you just set test vector to contain multiple of 8 characters?

Well, this is a test and tests should use unusual length and pass. This shows that something is not right and should be fixed. I only submit to show to everyone the problem. If someone have a tip I can check and send the fix.
On my view there is no point to workaround a problem on a test.

Reply:
> And by rounding the value to 40 just like that, aren't you actually crossing the tst_vec boundary and wandering into stack area that is not part of tst_vec (buffer overrun)?

Again, this is only to show the problem.

Reply:
Fine, but tests should test your solution, your code. If your code uncovers some problem in some other layer that is fine. As far as I can tell supporting non-aligned  `flash_write` is not crucial for neither your code nor your tests.